src/main/scala/chisel3/util/experimental/decode/TruthTable.scala

```diff
@@ -28,25 +28,12 @@ final class TruthTable(val table: Map[BitPat, BitPat], val default: BitPat) {
 }
 
 object TruthTable {
-  /** Parse TruthTable from its string representation. */
-  def apply(tableString: String): TruthTable = {
-    TruthTable(
-      tableString
-        .split("\n")
-        .filter(_.contains("->"))
-        .map(_.split("->").map(str => BitPat(s"b$str")))
-        .map(bps => bps(0) -> bps(1))
-        .toSeq,
-      BitPat(s"b${tableString.split("\n").filterNot(_.contains("->")).head.replace(" ", "")}")
-    )
-  }
-
   /** Convert a table and default output into a [[TruthTable]]. */
-  def apply(table: Iterable[(BitPat, BitPat)], default: BitPat): TruthTable = {
+  def apply(table: Iterable[(BitPat, BitPat)], default: BitPat, sorted: Boolean = true): TruthTable = {
     require(table.map(_._1.getWidth).toSet.size == 1, "input width not equal.")
     require(table.map(_._2.getWidth).toSet.size == 1, "output width not equal.")
     val outputWidth = table.map(_._2.getWidth).head
```

Comment:
Do we really want to trust the user has provided a sorted table? I think it might just be better to sort unconditionally.

Reply:
I added the default option sort=true
By default in the apply function it will sort user’s input. If user turn it off, which means user should take the unsorted risc(I should document this)

Reply:
Ah I see, yeah please document what it does. Also, I'd call it `sort` rather than `sorted` because the latter implies to me that the input is already sorted rather than that we will sort for you.

Reply:
sure ;p